File: articles/numerical-methods-and-algorithmic-approximation/haskell/NumericalApproximation.hs
module Main where

f :: Double -> Double
f x = sin x + 0.25 * x * x

centralDifference :: Double -> Double -> Double
centralDifference x h = (f (x + h) - f (x - h)) / (2 * h)

main :: IO ()
main = print (centralDifference 1.0 0.01)

File: articles/numerical-methods-and-algorithmic-approximation/typescript/numerical_approximation.ts
function f(x: number): number { return Math.sin(x) + 0.25 * x * x; }
function centralDifference(x: number, h: number): number { return (f(x + h) - f(x - h)) / (2 * h); }
console.log(centralDifference(1.0, 0.01).toFixed(12));
